fe2-android/app/src/main/java/com/github/dedis/popstellar/model/objects/Lao.java

```diff
@@ -271,20 +274,19 @@ public void setWitnesses(Set<PublicKey> witnesses) {
       }
     }
     this.witnesses = witnesses;
-    witnesses.forEach(
-        w -> {
-          if (nodes.stream().noneMatch(node -> node.getPublicKey().equals(w))) {
-            nodes.add(new ConsensusNode(w));
-          }
-        });
+    witnesses.forEach(w -> keyToNode.computeIfAbsent(w, ConsensusNode::new));
   }
 
   public void setPendingUpdates(Set<PendingUpdate> pendingUpdates) {
     this.pendingUpdates = pendingUpdates;
   }
 
   public List<ConsensusNode> getNodes() {
-    return nodes;
+    return new ArrayList<>(keyToNode.values());
+  }
```

Comment:
The list returned by getNodes() of Lao is not in a fixed order.
The order of nodes are not used anymore.
For example in the ElectionStartFragmentTest I get the position in the GridView by comparing the publicKey

Reply:
So, why do you return a list ?

Reply:
The previous code was using a list and I keep using a list to avoid more refactor.
It could be change to a Set but then it will need to be converted back to a list at some point.
The getNodes() is here only to get a list of all nodes (but not in any particular order). Do you think it could be better to return a sorted list even if it won't be used as sorted ? Or a Set and then convert it back to a list ?

Reply:
I changed the getNode() and it is now returning a sorted List based on the base64 kex representation.